app/Listeners/QueueMessage.php

```diff
@@ -35,16 +37,18 @@ public function handle(QueueMessageRequest $event)
         // Build the email.
         $email = new Email($resources);
 
-        foreach ($email->allMessages as $content) {
-            $settings = [
-                'subject' => $content['message']['subject'],
-                'from' => $email->contest->sender_email,
-                'from_name' => $email->contest->sender_name,
-                'to' => $content['user']->email,
-                'to_name' => $content['user']->first_name,
-            ];
+        if (isset($email->allMessages)) {
+            foreach ($email->allMessages as $content) {
+                $settings = [
+                    'subject' => $content['message']['subject'],
+                    'from' => $email->contest->sender_email,
+                    'from_name' => $email->contest->sender_name,
+                    'to' => $content['user']->email,
+                    'to_name' => $content['user']->first_name,
+                ];
 
-            $this->sendMail($content['message'], $settings);
+                $this->sendMail($content['message'], $settings);
```

Comment:
I'm still a bit confused by this, do welcome emails still send? 
if we are sending a one off email to just one user, don't we need to check [here](https://github.com/DoSomething/gladiator/blob/master/app/Services/Email.php#L154)?
won't this  `sendMail` call not fire because it's within the for loop if there are multiple messages?